osu.Game/Screens/Play/Player.cs

```diff
@@ -443,6 +443,11 @@ public void Pause()
         {
             if (!canPause) return;
 
+            if (IsResuming)
+            {
+                DrawableRuleset.CancelResume();
+            }
```

Comment:
Passing those lines of code without a comment makes it a bit confusing as to why it's there (or for me at least), please provide a comment as to why the resume overlay should be canceled here.

Reply:
Also, resuming check doesn't seem to be required, it's fine to hide a drawable that is already hidden. (nothing will change)

Reply:
I did add the commenting, but I tried to remove the check and it threw an error with normal pausing, so I had to keep it in.

Reply:
I had a look at why that fails - without the check `OsuResumeOverlay.Hide()` eats a `NullReferenceException` here at first pause:

https://github.com/ppy/osu/blob/50b674249ba921bbcf2a0145062639d62249c8c6/osu.Game.Rulesets.Osu/UI/OsuResumeOverlay.cs#L62

due to `GameplayCursor` being null. Unsure whether we want to guard for null there and remove this check, or just leave the check as-is here.

Reply:
I think it would be better to do a null check there, for no other reason than it saves a bit of space, and makes it a little more organized in the Pause method.

Reply:
I advocate for a null check as there's simply no reason for the resume overlay to be only hidden once.